### PR TITLE
e2e-node: support CPU of more than 3 thread in one core in CPUManagerTest

### LINKED ISSUE #118968: [Flaky test][sig-node] With kubeconfig updated to static CPU Manager policy run the Topology Manager tests run Topology Manager policy test suite
### Which jobs are flaking?

[sig-node] Topology Manager [Serial] [NodeFeature:TopologyManager] With kubeconfig updated to static CPU Manager policy run the Topology Manager tests run Topology Manager policy test suite

We saw the failure occasionally on arm64, pls see: https://github.com/kubernetes/kubernetes/pull/118919

> [sig-node] Topology Manager [Serial] [NodeFeature:TopologyManager] With kubeconfig updated to static CPU Manager policy run the Topology Manager tests run Topology Manager policy test suite
test/e2e_node/topology_manager_test.go:861
  STEP: Creating a kubernetes client @ 06/28/23 00:10:44.522
  STEP: Building a namespace api object, basename topology-manager-test @ 06/28/23 00:10:44.522
  Jun 28 00:10:44.524: INFO: Skipping waiting for service account
  STEP: by configuring Topology Manager policy to single-numa-node @ 06/28/23 00:10:44.534
  Jun 28 00:10:44.534: INFO: Configuring topology Manager policy to single-numa-node
  Jun 28 00:10:44.534: INFO: failed to find any VF device from [{0000:00:00.0 -1 false false} {0000:00:01.0 -1 false false} {0000:00:04.0 0 false false} {0000:00:05.0 -1 false false}]
  Jun 28 00:10:44.534: INFO: New kubelet config is {{ } %!s(bool=true) /tmp/node-e2e-20230627T232107/static-pods2504443847 {1m0s} {10s} {20s}  map[] 0.0.0.0 %!s(int32=10250) %!s(int32=10255) /usr/libexec/kubernetes/kubelet-plugins/volume/exec/  /var/lib/kubelet/pki/kubelet.crt /var/lib/kubelet/pki/kubelet.key []  %!s(bool=false) %!s(bool=false) {{} {%!s(bool=false) {2m0s}} {%!s(bool=true)}} {AlwaysAllow {{5m0s} {30s}}} %!s(int32=5) %!s(int32=10) %!s(int32=50) %!s(int32=100) %!s(bool=true) %!s(bool=false) %!s(int32=10248) 127.0.0.1 %!s(int32=-999) cluster.local [] {4h0m0s} {10s} {5m0s} %!s(int32=40) {2m0s} %!s(int32=85) %!s(int32=80) {10s} /kubelet.slice  / %!s(bool=true) systemd static map[] {1s} None single-numa-node container map[] map[] {2m0s} promiscuous-bridge %!s(int32=110) 10.100.0.0/24 %!s(int64=-1) /etc/resolv.conf %!s(bool=false) %!s(bool=true) {100ms} %!s(int64=1000000) %!s(int32=50) application/vnd.kubernetes.protobuf %!s(int32=50) %!s(int32=100) %!s(bool=false) %!s(*int32=<nil>) map[memory.available:250Mi nodefs.available:10% nodefs.inodesFree:5%] map[] map[] {30s} %!s(int32=0) map[nodefs.available:5% nodefs.inodesFree:5%] %!s(int32=0) %!s(bool=true) %!s(bool=false) %!s(bool=true) %!s(int32=14) %!s(int32=15) map[DisableKubeletCloudCredentialProviders:%!s(bool=true)] %!s(bool=true) {} 10Mi %!s(int32=5) Watch [] %!s(bool=true) map[] map[cpu:200m]   [pods]   {text 5s %!s(v1.VerbosityLevel=4) [] {{%!s(bool=false) {{{%!s(int64=0) %!s(resource.Scale=0)} {%!s(*inf.Dec=<nil>)} 0 DecimalSI}}}}} %!s(bool=true) %!s(bool=false) {0s} {0s} [] [] %!s(bool=true) %!s(bool=true) %!s(bool=false) %!s(*float64=0x4001977560) [] %!s(bool=true) %!s(*v1.TracingConfiguration=<nil>) %!s(bool=true) unix:///run/containerd/containerd.sock }
  STEP: Stopping the kubelet @ 06/28/23 00:10:44.534
  Jun 28 00:10:44.546: INFO: Get running kubelet with systemctl:   UNIT                            LOAD   ACTIVE SUB     DESCRIPTION
    kubelet-20230627T232107.service loaded active running /tmp/node-e2e-20230627T232107/kubelet --kubeconfig /tmp/node-e2e-20230627T232107/kubeconfig --root-dir /var/lib/kubelet --v 4 --hostname-override i-0f1d9bf08a330dd96 --container-runtime-endpoint unix:///run/containerd/containerd.sock --config /tmp/node-e2e-20230627T232107/kubelet-config --kernel-memcg-notification=true --feature-gates=DisableKubeletCloudCredentialProviders=true --image-credential-provider-config=/tmp/node-e2e-20230627T232107/credential-provider.yaml --image-credential-provider-bin-dir=/tmp/node-e2e-20230627T232107 --cluster-domain=cluster.local --cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/containerd.service

>  LOAD   = Reflects whether the unit definition was properly loaded.
  ACTIVE = The high-level unit activation state, i.e. generalization of SUB.
  SUB    = The low-level unit activation state, values depend on unit type.
  1 loaded units listed.
  , kubelet-20230627T232107
  W0628 00:10:44.598698    2723 util.go:496] Health check on "http://127.0.0.1:10248/healthz" failed, error=Head "http://127.0.0.1:10248/healthz": read tcp 127.0.0.1:59126->127.0.0.1:10248: read: connection reset by peer
  STEP: Starting the kubelet @ 06/28/23 00:10:44.603
  W0628 00:10:44.632245    2723 util.go:496] Health check on "http://127.0.0.1:10248/healthz" failed, error=Head "http://127.0.0.1:10248/healthz": dial tcp 127.0.0.1:10248: connect: connection refused
  STEP: checking node CPU capacity (4) and allocatable CPUs (3) @ 06/28/23 00:10:49.638
  STEP: running a non-Gu pod @ 06/28/23 00:10:49.638
  STEP: checking if the expected cpuset was assigned @ 06/28/23 00:10:51.65
  Jun 28 00:10:51.660: INFO: Unexpected error: expected log not found in container [non-gu-container] of pod [non-gu-pod]: 
      <*errors.errorString | 0x40014038d0>: 
      failed to match regexp "^0-3\n$" in output "0,3\n"
      {
          s: "failed to match regexp \"^0-3\\n$\" in output \"0,3\\n\"",
      }
  [FAILED] in [It] - test/e2e_node/cpu_manager_test.go:275 @ 06/28/23 00:10:51.661

A different error msg with arm64  on ubuntu18.04

> [FAILED] expected log not found in container [gu-container] of pod [gu-pod]: failed to match regexp "^1,29,57,85\n$" in output "28,56\n"
In [It] at: /go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/cpu_manager_test.go:388 @ 06/28/23 14:35:53.953

### Which tests are flaking?

[sig-node] Topology Manager [Serial] [NodeFeature:TopologyManager] With kubeconfig updated to static CPU Manager policy run the Topology Manager tests run Topology Manager policy test suite

### Since when has it been flaking?

the could be found occasionally on arm64, and I double the issue exists for quite a long time.

### Testgrid link

_No response_

### Reason for failure (if possible)

_No response_

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig node

### PR DESCRIPTION
**Pull request by ruquanzhao**
#### What type of PR is this?
/kind bug
/kind flake

#### What this PR does / why we need it:
The current /test/e2e_node/cpu_manager_test.go don't consider CPUs that there are more than 3 threads in one core.

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/118968

#### Does this PR introduce a user-facing change?
NONE

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ffromani** [comment]
/lgtm

the current code could use some attention because it really feels unnecessarily hard to follow. This PR seems to fix the issue so it's a welcome step to improve the noise and enabling us a rewrite later.

**Comment by ruquanzhao** [comment]
the job 

- pull-kubernetes-node-kubelet-serial-cpu-manager
- pull-kubernetes-node-kubelet-serial-cpu-manager-kubetest2
- ci-kubernetes-node-kubelet-serial-cpu-manager

is unhealthy, will look into it.

**Comment by ffromani** [comment]
> the job
> 
>     * pull-kubernetes-node-kubelet-serial-cpu-manager
> 
>     * pull-kubernetes-node-kubelet-serial-cpu-manager-kubetest2
> 
>     * ci-kubernetes-node-kubelet-serial-cpu-manager
> 
> 
> is unhealthy, will look into it.

Thanks @ruquanzhao . We are looking into it already. It's unfortunate we have unhealthy jobs now and gets in the way of this PR, but since the whole point of this PR is improving the e2e tests, we need to have some evidence the change is fixing. If there are other options to prove that (code looks good btw) I'll be happy to consider them.

**Comment by ffromani** [comment]
xref: https://github.com/kubernetes/kubernetes/issues/118851

**Comment by ruquanzhao** [comment]
> we need to have some evidence the change is fixing.

Hi, @ffromani, since we don't have test environments that there is four threads in one core, this is hard to provide evidence.
Since this pr is for very special circumstances, how about we ignore this and just print some hints in e2e-node?
